### PR TITLE
first homework submission

### PR DESCRIPTION
**Pull request by kevin-hw-li**
Difficulty: 7.5

Comment: 
- Took quite some time to make the Word Guesser works.
- Have trouble with the strings exercise FixStart.

### CODE REVIEW

**Comment by kevin-hw-li** [comment]
Added second homework submission.

Difficulty: 7

File: kevin_li/week01/day3/word_guesser/js/main.js [hunk 1]
```diff
@@ -15,25 +15,34 @@ var guess = ['_', '_', '_'];  // line 5
...
-    } else if (guess[i] == '_') {
-      console.log('Sadness.');
-    } else if (guess[i] != '_') {
-      console.log('YOU HAVE WON!');
+      var correct = true;
     }
   }
-}
+  if (correct) {
+    if (guess[0] != '_' && guess[1] != '_' && guess[2] != '_') {  // line 11
```
Comment (textchimp) [inline kevin_li/week01/day3/word_guesser/js/main.js]: Interesting way to test for a win!

I still think a counter variable to count the correct guesses would be easier, i.e. `if(correctGuesses === word.length)`

File: kevin_li/week01/day4/javascript_bank/js/main.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  }
+};
+
+deposit('Dominique', 1000)
+
+
+// Withdrawal
+var withdrawal = function (name, withdrawAmount) {
+  for (var i = 0; i < bankAccounts.length; i++) {
+    if (name === bankAccounts[i].name && bankAccounts[i].balance < withdrawAmount) {
```
Comment (textchimp) [inline kevin_li/week01/day4/javascript_bank/js/main.js:82]: I'd suggest making a helper function to give you the index of an account name in the accounts array, so you don't always have to do this looping and testing for account names whenever you make a transaction.

File: kevin_li/week01/day5/MTA/js/main.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+// Line name
+var lineName = function (line) {
+  if (line === lineN) {
+    return 'N line';
+  } else if (line === lineL) {
+    return 'L line';
+  } else if (line === line6) {
+    return '6 line';
+  }
```
Comment (textchimp) [inline kevin_li/week01/day5/MTA/js/main.js:77]: If you used an object as your main data structure, with the line names as keys, and the array of stations on the line as the key's value, you wouldn't have to do all this translation of variable names into strings.... much easier.

**Comment by textchimp** [comment]
Looks like a good MTA attempt! See my previous comment about using an object as your main data structure to simplify things. 

Also, I'd say you have too many mysterious single-letter variable names like `i`, `j`, `k` etc. These are okay when used in a very limited and conventional way, i.e. as the loop counter in `for` loops, but should otherwise be avoided in favour of more descriptive and readable names, like `startIndex` and so on.

**Comment by kevin-hw-li** [comment]
Thanks!
I will work on simplifying the codes and make them better.

**Pull request merged**